src/Sylius/Component/Product/Model/Product.php

```diff
@@ -288,10 +306,15 @@ public function hasAttribute(AttributeValueInterface $attribute)
     /**
      * {@inheritdoc}
      */
-    public function hasAttributeByCode($attributeCode)
+    public function hasAttributeByCodeAndLocale($attributeCode, $localeCode = null)
     {
+        if (null === $localeCode) {
+            $localeCode = $this->getTranslation()->getLocale();
+        }
+
         foreach ($this->attributes as $attribute) {
-            if ($attribute->getAttribute()->getCode() === $attributeCode) {
+            if ($attribute->getAttribute()->getCode() === $attributeCode &&
+                $attribute->getLocaleCode() === $localeCode) {
                 return true;
```

Comment:
Didn't notice that the if above has `and`... My bad. Anyway, the if above will fit one line. And also we have to decide for some convention with multiple conditions in on if statement. Here you have `&&` at the end of first line, and a few lines below the `||` is at the beginning of a second one...